### ISSUE TITLE
[1.18.1] Issue with JourneyMapPlugin

### ISSUE THREAD

**Issue by itsdinkd** [ldtteam/minecolonies#7912]
Checking logs and am seeing the journeymap plugin fail for minecolonies. 

```java
[19:25:37] [Render thread/ERROR]: Plugin JourneymapPlugin{modId=minecolonies} errored during event: MAPPING_STOPPED
java.lang.NoSuchFieldError: level
	at com.minecolonies.coremod.compatibility.journeymap.JourneymapPlugin.onEvent(JourneymapPlugin.java:49) ~[minecolonies-1.18.1-1.0.328-ALPHA.jar%23238!:1.18.1-1.0.328-ALPHA]
	at journeymap.client.api.impl.PluginWrapper.notify(PluginWrapper.java:293) ~[journeymap-1.18.1-5.8.0beta5.jar%23210!:1.18.1-5.8.0beta5]
	at journeymap.client.api.impl.ClientEventManager.fireMappingEvent(ClientEventManager.java:76) ~[journeymap-1.18.1-5.8.0beta5.jar%23210!:1.18.1-5.8.0beta5]
	at journeymap.client.JourneymapClient.stopMapping(JourneymapClient.java:600) ~[journeymap-1.18.1-5.8.0beta5.jar%23210!:1.18.1-5.8.0beta5]
	at journeymap.client.forge.event.WorldEventHandler.onUnload(WorldEventHandler.java:37) ~[journeymap-1.18.1-5.8.0beta5.jar%23210!:1.18.1-5.8.0beta5]
	at net.minecraftforge.eventbus.ASMEventHandler_2721_WorldEventHandler_onUnload_Unload.invoke(.dynamic) ~[?:?]
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85) ~[eventbus-5.0.3.jar%232!:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-5.0.3.jar%232!:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-5.0.3.jar%232!:?]
	at net.minecraft.client.Minecraft.m_91320_(Minecraft.java:2120) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.client.gui.screens.PauseScreen.m_96314_(PauseScreen.java:77) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.client.gui.components.Button.m_5691_(Button.java:40) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.client.gui.components.AbstractButton.m_5716_(AbstractButton.java:16) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.client.gui.components.AbstractWidget.m_6375_(AbstractWidget.java:111) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.client.gui.components.events.ContainerEventHandler.m_6375_(ContainerEventHandler.java:27) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.client.MouseHandler.lambda$onPress$0(MouseHandler.java:88) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.client.gui.screens.Screen.m_96579_(Screen.java:527) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.client.MouseHandler.m_91530_(MouseHandler.java:85) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.client.MouseHandler.m_168091_(MouseHandler.java:185) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.util.thread.BlockableEventLoop.execute(BlockableEventLoop.java:101) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.client.MouseHandler.m_91565_(MouseHandler.java:184) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:36) ~[lwjgl-glfw-3.2.2.jar%2348!:build 10]
	at org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.2.2.jar%2344!:build 10]
	at org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3101) ~[lwjgl-glfw-3.2.2.jar%2348!:build 10]
	at com.mojang.blaze3d.systems.RenderSystem.m_69495_(RenderSystem.java:195) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at com.mojang.blaze3d.platform.Window.m_85435_(Window.java:333) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.client.Minecraft.m_91383_(Minecraft.java:1062) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.client.Minecraft.m_91374_(Minecraft.java:660) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at fudge.notenoughcrashes.mixinhandlers.InGameCatcher.handleClientCrash(InGameCatcher.java:37) ~[notenoughcrashes-4.1.3+1.18.1-forge.jar%23259!:?]
	at net.minecraft.client.Minecraft.modify$zmh000$atTheEndOfFirstCatchBeforePrintingCrashReport(Minecraft.java:6392) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.client.Minecraft.m_91374_(Minecraft.java:680) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.client.main.Main.main(Main.java:205) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.minecraftforge.fml.loading.targets.CommonClientLaunchHandler.lambda$launchService$0(CommonClientLaunchHandler.java:45) ~[fmlloader-1.18.1-39.0.10.jar%2323!:?]
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-9.0.7.jar%235!:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) [modlauncher-9.0.7.jar%235!:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) [modlauncher-9.0.7.jar%235!:?]
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:106) [modlauncher-9.0.7.jar%235!:?]
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:77) [modlauncher-9.0.7.jar%235!:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) [modlauncher-9.0.7.jar%235!:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) [modlauncher-9.0.7.jar%235!:?]
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:90) [bootstraplauncher-0.1.17.jar:?]
[19:25:38] [Server thread/ERROR]: Plugin JourneymapPlugin{modId=minecolonies} errored during event: MAPPING_STOPPED
java.lang.NoSuchFieldError: level
	at com.minecolonies.coremod.compatibility.journeymap.JourneymapPlugin.onEvent(JourneymapPlugin.java:49) ~[minecolonies-1.18.1-1.0.328-ALPHA.jar%23238!:1.18.1-1.0.328-ALPHA]
	at journeymap.client.api.impl.PluginWrapper.notify(PluginWrapper.java:293) ~[journeymap-1.18.1-5.8.0beta5.jar%23210!:1.18.1-5.8.0beta5]
	at journeymap.client.api.impl.ClientEventManager.fireMappingEvent(ClientEventManager.java:76) ~[journeymap-1.18.1-5.8.0beta5.jar%23210!:1.18.1-5.8.0beta5]
	at journeymap.client.JourneymapClient.stopMapping(JourneymapClient.java:600) ~[journeymap-1.18.1-5.8.0beta5.jar%23210!:1.18.1-5.8.0beta5]
	at journeymap.client.forge.event.WorldEventHandler.onUnload(WorldEventHandler.java:37) ~[journeymap-1.18.1-5.8.0beta5.jar%23210!:1.18.1-5.8.0beta5]
	at net.minecraftforge.eventbus.ASMEventHandler_2721_WorldEventHandler_onUnload_Unload.invoke(.dynamic) ~[?:?]
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85) ~[eventbus-5.0.3.jar%232!:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-5.0.3.jar%232!:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-5.0.3.jar%232!:?]
	at net.minecraft.server.MinecraftServer.m_7041_(MinecraftServer.java:599) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.client.server.IntegratedServer.m_7041_(IntegratedServer.java:176) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:706) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.server.MinecraftServer.m_177918_(MinecraftServer.java:258) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
[19:25:38] [Server thread/ERROR]: Plugin JourneymapPlugin{modId=minecolonies} errored during event: MAPPING_STOPPED
java.lang.NoSuchFieldError: level
	at com.minecolonies.coremod.compatibility.journeymap.JourneymapPlugin.onEvent(JourneymapPlugin.java:49) ~[minecolonies-1.18.1-1.0.328-ALPHA.jar%23238!:1.18.1-1.0.328-ALPHA]
	at journeymap.client.api.impl.PluginWrapper.notify(PluginWrapper.java:293) ~[journeymap-1.18.1-5.8.0beta5.jar%23210!:1.18.1-5.8.0beta5]
	at journeymap.client.api.impl.ClientEventManager.fireMappingEvent(ClientEventManager.java:76) ~[journeymap-1.18.1-5.8.0beta5.jar%23210!:1.18.1-5.8.0beta5]
	at journeymap.client.JourneymapClient.stopMapping(JourneymapClient.java:600) ~[journeymap-1.18.1-5.8.0beta5.jar%23210!:1.18.1-5.8.0beta5]
	at journeymap.client.forge.event.WorldEventHandler.onUnload(WorldEventHandler.java:37) ~[journeymap-1.18.1-5.8.0beta5.jar%23210!:1.18.1-5.8.0beta5]
	at net.minecraftforge.eventbus.ASMEventHandler_2721_WorldEventHandler_onUnload_Unload.invoke(.dynamic) ~[?:?]
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85) ~[eventbus-5.0.3.jar%232!:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-5.0.3.jar%232!:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-5.0.3.jar%232!:?]
	at net.minecraft.server.MinecraftServer.m_7041_(MinecraftServer.java:599) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.client.server.IntegratedServer.m_7041_(IntegratedServer.java:176) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:706) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at net.minecraft.server.MinecraftServer.m_177918_(MinecraftServer.java:258) ~[client-1.18.1-20211210.034407-srg.jar%23346!:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]

```

**Comment by Raycoms** [comment]
Are you on the latest journeymap version?

**Comment by uecasm** [comment]
Actually there's a bit of an API break in JM that is contributing to this problem.

Minecolonies 328 uses the JM 1.7 API which is only supported by exactly JM 5.8.0beta4 (the API itself also supports older versions, but there are bugs in older versions of JM that are not compatible with mcol).

JM 5.8.0beta5 uses JM API 1.8 instead, which includes this particular breaking change among other things.  A patch to support this new API has already been made to mcol but for some unknown reason there isn't yet a 1.18 build that includes this (even though it was merged).

So for now, you will need to downgrade JM to beta4.  Once the next mcol 1.18 update is available you should be able to update to it and then upgrade to beta5 or beta6.

**Comment by uecasm** [comment]
FWIW, this particular API break also affects 1.17 (where a new build is similarly not available yet), but does not affect 1.16 (where a new build is available, but also a mixture of builds will work as well).

**Comment by Raycoms** [comment]
Build pushed that works
